.github/workflows/ci.yaml

```
@@ -0,0 +1,26 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - "main"
+  push:
+    branches:
+      - "main"
```

Nit, recommend using these triggers. The main difference is running on all PRs (sometimes we'll merge one branch into another), and ability to manually dispatch the workflow if needed
```suggestion
on:
  workflow_dispatch:
  pull_request:
  push:
    branches:
      - main
```